src/Symfony/Bridge/Monolog/Handler/ElasticsearchLogstashHandler.php

```diff
@@ -63,7 +63,7 @@ class ElasticsearchLogstashHandler extends AbstractHandler
 
     public function __construct(string $endpoint = 'http://127.0.0.1:9200', string $index = 'monolog', ?HttpClientInterface $client = null, string|int|Level $level = Logger::DEBUG, bool $bubble = true, string $elasticsearchVersion = '1.0.0')
     {
-        if (!interface_exists(HttpClientInterface::class)) {
+        if (!interface_exists(HttpClientInterface::class) || !$client && !class_exists(HttpClient::class)) {
```

Comment:
```suggestion
        if (!$client && !class_exists(HttpClient::class)) {
```

There is no need to check for the interface anymore.

Reply:
Good point! Indeed, if we receive an object, it is necessarily an instance of the interface.